wgengine/magicsock/magicsock.go

```diff
@@ -5090,40 +5090,61 @@ func betterAddr(a, b addrLatency) bool {
 	if !a.IsValid() {
 		return false
 	}
-	if a.Addr().Is6() && b.Addr().Is4() {
-		// Prefer IPv6 for being a bit more robust, as long as
-		// the latencies are roughly equivalent.
-		if a.latency/10*9 < b.latency {
-			return true
-		}
-	} else if a.Addr().Is4() && b.Addr().Is6() {
-		if betterAddr(b, a) {
-			return false
-		}
+
+	// Each address starts with a set of points (from 0 to 100) that
+	// represents how much faster they are than the highest-latency
+	// endpoint. For example, if a has latency 200ms and b has latency
+	// 190ms, then a starts with 0 points and b starts with 5 points since
+	// it's 5% faster.
+	highestLatency := a.latency
+	if b.latency > highestLatency {
+		highestLatency = b.latency
+	}
+	var aPoints, bPoints int
+	if highestLatency > 0 {
+		aPoints = int(100 - ((a.latency * 100) / highestLatency))
+		bPoints = int(100 - ((b.latency * 100) / highestLatency))
```

Comment:
One of these will always be 0, the other will be between 0 and 100. It's probably going to be simpler just to do
```golang
if a.latency > b.latency && a.latency > 0 {
   bPoints = int(100 - ((b.latency * 100) / a.latency))
} else if b.latency > 0 {
   aPoints = int(100 - ((a.latency * 100) / b.latency))
}
```